pkg/smart_wallet/factory_calldata.go

```diff
@@ -0,0 +1,70 @@
+package smart_wallet
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/shopspring/decimal"
+)
+
+func GetFactoryCallData(smartWalletConfig Config, ownerAddress common.Address, index decimal.Decimal) ([]byte, error) {
```

Comment:
I recommend adding a godoc comment here to provide context and explain the why of this function.

Reply:
6982100528d7e2cfb6f19798fbceeaf3e24497b1